ios/BrightcovePlayer.m

```diff
@@ -129,20 +143,126 @@ - (void)playbackController:(id<BCOVPlaybackController>)controller playbackSessio
         if (self.onReady) {
             self.onReady(@{});
         }
+        if (self.onStatusEvent) {
+            self.onStatusEvent(@{
+                                 @"type": @("ready")
+                                 });
+        }
     } else if (lifecycleEvent.eventType == kBCOVPlaybackSessionLifecycleEventPlay) {
         _playing = true;
         if (self.onPlay) {
             self.onPlay(@{});
         }
+        if (self.onStatusEvent) {
+            self.onStatusEvent(@{
+                                 @"type": @("play")
+                                 });
+        }
     } else if (lifecycleEvent.eventType == kBCOVPlaybackSessionLifecycleEventPause) {
         _playing = false;
         if (self.onPause) {
             self.onPause(@{});
         }
+        if (self.onStatusEvent) {
+            self.onStatusEvent(@{
+                                 @"type": @("pause")
+                                 });
+        }
     } else if (lifecycleEvent.eventType == kBCOVPlaybackSessionLifecycleEventEnd) {
         if (self.onEnd) {
             self.onEnd(@{});
         }
+        if (self.onStatusEvent) {
+            self.onStatusEvent(@{
+                                 @"type": @("end")
+                                 });
+        }
+    } else if (lifecycleEvent.eventType == kBCOVPlaybackSessionLifecycleEventFail) {
+        if (self.onStatusEvent) {
+            
+            NSString* error = @"";
```

Comment:
Can you try doing `NSString* error = (NSString*)[NSNull null]'`

Reply:
or change it to `NSString* error = nil;` and then for adding in the key do `@"error": (error ? error : [NSNull null])`